### PR TITLE
Updated code

### PR DESCRIPTION
**Pull request by romulogcerqueira**
- update sonar simulators with new rock_widget_collection viewers
- optimized code
- solve the "black wholes" on simulated sonar image

### CODE REVIEW

File: src/CommonSonar.cpp [hunk 1]
```diff
@@ -20,48 +20,63 @@ float CommonSonar::sigmoid(float x) {
...
-
-	float interval = 1.0 / _number_of_bins * 1.0;
-
-	cv::Mat bins_depth;
-	cv::Mat bins_normal = cv::Mat::zeros(_number_of_bins, 1, CV_32F);
+std::vector<double> CommonSonar::decodeShaderImage(cv::Mat raw_image) {
+
+    if (raw_image.type() != CV_32FC3) {
+        std::cout << "Invalid shader image format!" << std::endl;
+        exit(0);
```
Comment (doudou) [inline src/CommonSonar.cpp]: Never
Do
That.

Throw an exception

File: src/CommonSonar.hpp [hunk 2]
```diff
@@ -29,15 +29,15 @@ class CommonSonar {
...
 		_speed_of_sound(1500.0f),
 		_range(50.0f),
 		_beamwidth_horizontal(0.0f),
 		_beamwidth_vertical(0.0f)
 	{};
 
-	cv::Mat decodeShaderImage(cv::Mat raw_image);
-	std::vector<uint8_t> getPingData(cv::Mat raw_intensity);
+	std::vector<double> decodeShaderImage(cv::Mat raw_image);
+	std::vector<uint8_t> getPingData(std::vector<double> raw_intensity);
```
Comment (doudou) [inline src/CommonSonar.hpp]: cv::Mat and std::vector are better passed as const-references.

Comment (romulogcerqueira) [inline src/CommonSonar.hpp]: Done!

File: src/CommonSonar.cpp (hunk 1 above)
Comment (romulogcerqueira) [inline src/CommonSonar.cpp]: Done!

File: src/CommonSonar.cpp [hunk 3]
```diff
@@ -6,7 +6,7 @@ using namespace gpu_sonar_simulation;
 namespace gpu_sonar_simulation {
 
 // Accepts the input value x then returns it's sigmoid value in float
-float CommonSonar::sigmoid(float x) {
+float CommonSonar::sigmoid(const float& x) {
```
Comment (doudou) [inline src/CommonSonar.cpp]: The C++ default idiom is to use const-ref for any complex type (struct, class) and value for fundamental ones (numeric, pointers, enums)

File: src/CommonSonar.cpp [hunk 4]
```diff
@@ -20,48 +20,61 @@ float CommonSonar::sigmoid(float x) {
...
+        }
+    }
 
-	return bins_normal;
+    return new_hist;
 }
 
 // Calculate ping intensity in 8-bit format data
-std::vector<uint8_t> CommonSonar::getPingData(cv::Mat raw_intensity) {
+std::vector<uint8_t> CommonSonar::getPingData(std::vector<double> raw_intensity) {
```
Comment (doudou) [inline src/CommonSonar.cpp]: const-ref

File: src/CommonSonar.cpp (hunk 3 above)
Comment (romulogcerqueira) [inline src/CommonSonar.cpp]: I got your point. Done!

File: src/CommonSonar.cpp (hunk 4 above)
Comment (romulogcerqueira) [inline src/CommonSonar.cpp]: Hi @doudou,

in this case, I change the raw_intensity values inside this function. As far as I know, I cannot pass this structure as a const type. I updated the code to pass a reference of raw_intensity's header and avoid copy.

**Pull request merged**